.circleci/config.yml

```diff
@@ -389,6 +389,14 @@ commands:
 jobs:
   spm-release-build-xcode-14:
     <<: *base-job
+    parameters:
+      xcode_version:
+        type: string
+        default: '14.3.0'
+      install_swiftlint:
+        type: boolean
+        default: false
```

Comment:
we were repeating the code versions every time we called the jobs. This could make sense for some jobs if you want to call it again with a different xcode version, but I couldn't find any where we seemed to be doing that intentionally. 

Also some of them (like this one) are even named after their xcode version, so it makes no sense to have it be an option. 

We could remove jobs that are exactly the same but only have different xcode versions and use the param instead, but that's for another day. Right now I'm trying to preserve logic as much as possible. 

So I just moved the xcode version to the jobs themselves.